### PR TITLE
chore: add `ef` as query parameter

### PR DESCRIPTION
**Pull request by BubbleCal**

### CODE REVIEW

File: rust/lance-index/src/vector.rs [hunk 1]
```diff
@@ -55,6 +55,10 @@ pub struct Query {
     /// The number of probes to load and search.
     pub nprobes: usize,
 
+    /// The number of candidates to reserve while searching.
+    /// this is an optional parameter for HNSW related index types.
+    pub ef: Option<usize>,
```
Comment (westonpace) [inline rust/lance-index/src/vector.rs:60]: Why `ef`?  I assume it is from the literature but is it short for something?

**Review by westonpace — APPROVED** [review]
If I read this PR correctly there is no way to actually specify `ef` from the user right?  I assume that will come later?

File: rust/lance-index/src/vector.rs (hunk 1 above)
Comment (BubbleCal) [reply rust/lance-index/src/vector.rs:60]: It stands for "explore factor", 'ef' may be more common

**Comment by BubbleCal** [comment]
> If I read this PR correctly there is no way to actually specify `ef` from the user right?  I assume that will come later?

I missed that modification, will add it in this PR

**Pull request merged**